src/v/cluster/controller_stm.cc

```diff
@@ -62,7 +62,14 @@ controller_stm::maybe_make_snapshot(ssx::semaphore_units apply_mtx_holder) {
     }
 
     controller_snapshot data;
-    // TODO: fill snapshot
+    ss::future<> fill_fut = ss::now();
+    auto call_stm_fill = [&fill_fut, &data](auto& stm) {
+        fill_fut = std::move(fill_fut).then(
```

Comment:
> why do we std::move(fill_fut) here? Its an lvalue?

I added std::move because `.then()` consumes the future. But I guess it is not needed because `.then()` accepts an lvalue. And static analysis didn't like it as well (erroneously, imo) so I'll get rid of it :)

> using a tuple here means we cannot use a parallel construct (like ss::parallel_for_each()) to fill_snapshot? std::apply here is serially processing the stms and we hold the mutex for longer.

See a [related discussion](https://github.com/redpanda-data/redpanda/pull/9601#discussion_r1147907199). Filling in parallel is possible, but it won't make anything faster because it is cpu-bound, and it is harder to understand.